pkgs/development/libraries/openssl/default.nix

```diff
@@ -122,6 +123,19 @@ let
       "MANSUFFIX=ssl"
     ];
 
+    buildFlags = lib.optionals static [
+      # Even though engines are disabled in static builds, we have to
+      # override ENGINESDIR so the bin output doesn't end up with an
+      # reference to the lib output.
+      "ENGINESDIR=/"
+    ];
+
+    installFlags = lib.optionals static [
+      # Build system wants to be able to create the engines directory
+      # even though nothing will get installed to it.
+      "ENGINESDIR=/build/engines"
```

Comment:
This should utilize `NIX_BUILD_TOP` instead of `/build` or we risk writing to the fs lacking the sandbox.